File: python/appendix_2_5_problem2_fine_depth_156_to_160.py
from __future__ import annotations

from x6 import x6


def run():
    """Appendix 2.5: problem 2, d = 156:0.01:160."""
    return x6()


if __name__ == "__main__":
    d, I, peak, peak_d = run()
    print("d =", d)
    print("I =", I)
    print("peak =", peak)
    print("peak_d =", peak_d)

File: matlab/appendix_2_5_problem2_fine_depth_156_to_160.m
function [d,I,peak,peak_d]=appendix_2_5_problem2_fine_depth_156_to_160
% 附录2.5：第二问精搜，d = 156:0.01:160 的分段积分计算。
[d,I]=x6;
[peak,i]=max(I);
peak_d=d(i);
